cmd/kube-apiserver/app/options/validation.go

```diff
@@ -39,6 +41,36 @@ func validateClusterIPFlags(options *ServerRunOptions) []error {
 		errs = append(errs, errors.New("specified --service-cluster-ip-range is too large"))
 	}
 
+	// Secondary IP validation
+	secondaryServiceClusterIPRangeUsed := (options.SecondaryServiceClusterIPRange.IP != nil)
+	if secondaryServiceClusterIPRangeUsed && !utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) {
+		errs = append(errs, fmt.Errorf("--secondary-service-cluster-ip-range can only be used if %v feature is enabled", string(features.IPv6DualStack)))
+	}
+
+	// note: While the cluster might be dualstack (i.e. pods with multiple IPs), the user may choose
+	// to only ingress traffic within and into the cluster on one IP family only. this family is decided
+	// by the range set on --service-cluster-ip-range. If/when the user decides to use dual stack services
+	// the Secondary* must be of different IPFamily than --service-cluster-ip-range
+	if secondaryServiceClusterIPRangeUsed {
```

Comment:
This check doesn't need a feature gate check?

Reply:
This https://github.com/kubernetes/kubernetes/pull/79386/commits/6521a8793ce8820f20c4c5f56642e6e5289eef41#diff-6696b3d19b862aab51fa1646fc7811abR46?